src/pykx/pandas_api/pandas_meta.py

```diff
@@ -255,3 +255,12 @@ def sum(self, axis=0, skipna=True, numeric_only=False, min_count=0):
             res,
             min_count
         ), cols)
+    
+    @convert_result
+    def nunique(self, axis=0, dropna=True):
+        res, cols = preparse_computations(self, axis, skipna=False)
+        filterNan = q('{$[11h = type x;x;'
+                 '0h = type x;(x where not null x except w),(w:x where 10h=type each x);'
+                 'x where not null x]}each')
+        res = filterNan(res) if dropna else res
+        return q('(\'[count;distinct]\')', res), cols
```

Comment:
I leave this alternative syntax in case someone finds it useful to understand Christian's incredible composition:

``` python
q("(count distinct @) each", res)
```